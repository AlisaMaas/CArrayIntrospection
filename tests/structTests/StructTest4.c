/**
* Ensure that array accesses don't
* fool us into adding incorrect results.
**/

struct foo {
	int a;
	double b;
	struct foo *c;
};

struct bar {
	struct foo first;
	int x;
	double y;
	struct bar *z;
};

void accessArrays(struct foo *f, struct bar *b) {
	for (int i = 0; i < 40; i++) {
		f[i] = f[0];
	}
	f[2] = f[0];
	b[200] = b[0];
	b[0] = b[0];
	b[1] = b[0];
	
}