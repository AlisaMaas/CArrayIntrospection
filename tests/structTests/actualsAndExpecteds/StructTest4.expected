Printing analysis 'Promote Memory to Register' for function 'accessArrays':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Organize GEP instructions referring to struct element accesses by struct element.':
